python/ray/air/_internal/usage.py

```diff
@@ -0,0 +1,299 @@
+            result = "custom_class_trainable"
+        else:
+            from ray.tune.experiment import Experiment
+
+            if isinstance(trainable, Experiment):
+                result = "experiment"
+            elif (
+                isinstance(trainable, Sequence)
+                and len(trainable) > 0
+                and isinstance(trainable[0], Experiment)
+            ):
+                result = "experiments" if len(trainable) > 1 else "experiment"
+            else:
+                result = "other"
+    _air_scenario.trainable = result
+
+
+# intended to be called only once upon entering `tune.run()`.
+def parse_and_set_search_algorithm(
+    algorithm: Optional[Union["Searcher", "SearchAlgorithm", str, "ConcurrencyLimiter"]]
+):
+    if algorithm is None:
+        result = "basic_variant"
+
+    from ray.tune.search import ConcurrencyLimiter
+    from ray.tune.search import SearchAlgorithm, Searcher
+
+    if isinstance(algorithm, ConcurrencyLimiter):
+        algorithm = algorithm.searcher
+    if isinstance(algorithm, str):
+        from ray.tune.search import SEARCH_ALG_IMPORT
+
+        if algorithm in SEARCH_ALG_IMPORT:
+            result = SEARCH_ALG_IMPORT[algorithm].__class__.__name__
+        else:
+            result = "invalid"
+    elif isinstance(algorithm, Searcher):
+        if algorithm.__module__.startswith("ray.tune.search"):
+            result = algorithm.__class__.__name__
+        else:
+            # We don't want to expose custom search/scheduler functions.
+            result = "custom"
+    elif isinstance(algorithm, SearchAlgorithm):
+        from ray.tune.search.basic_variant import BasicVariantGenerator
+
+        if isinstance(algorithm, BasicVariantGenerator):
+            result = "basic_variant"
+        else:
+            result = "custom"
+    else:
+        result = "other"
+    _air_scenario.search_alg = result
+
+
+# intended to be called only once upon entering `tune.run()`.
+def parse_and_set_scheduler_algorithm(
+    scheduler: Optional[Union[str, "TrialScheduler"]]
+):
+    from ray.tune.schedulers import TrialScheduler
+
+    if scheduler is None:
+        result = "none"
+    elif isinstance(scheduler, str):
+        from ray.tune.schedulers import SCHEDULER_IMPORT
+
+        if scheduler in SCHEDULER_IMPORT:
+            result = SCHEDULER_IMPORT[scheduler].__class__.__name__
+        else:
+            result = "invalid"
+    elif isinstance(scheduler, TrialScheduler):
+        if scheduler.__module__.startswith("ray.tune.schedulers"):
+            result = scheduler.__class__.__name__
+        else:
+            # We don't want to expose custom search/scheduler functions.
+            result = "custom_scheduler"
+    else:
+        result = "other"
+    _air_scenario.scheduler_alg = result
+
+
+# intended to be called when search algorithm finishes setting up.
+def set_more_than_one_trial(num_samples: int):
+    _air_scenario.more_than_one_trial = num_samples > 1
```

Comment:
Could we instead just store the number of trials, rather than this boolean?  Seems like we can use it for other things as well in the future, e.g. directly recording it as part of telemetry.